quantum/q_full_adder.py

```diff
@@ -0,0 +1,86 @@
+"""
+Build the quantum full adder (QFA) for any sum of
+two quantum registers and one carry in. This circuit
+is designed using the Qiskit framework. This
+experiment run in IBM Q simulator with 1000 shots.
+.
+References:
+https://www.quantum-inspire.com/kbase/full-adder/
+"""
+
+import qiskit
+from qiskit import Aer, ClassicalRegister, QuantumCircuit, QuantumRegister, execute
+
+
+def quantum_full_adder(
+    input_1: int = 1, input_2: int = 1, carry_in: int = 1
+) -> qiskit.result.counts.Counts:
+    """
+    # >>> q_full_adder(inp_1, inp_2, cin)
+    # the inputs can be 0/1 for qubits in define
+    # values, or can be in a superposition of both
+    # states with hadamard gate using the input value 2.
+    # result for default values: {11: 1000}
+    qr_0: ──■────■──────────────■──
+            │  ┌─┴─┐          ┌─┴─┐
+    qr_1: ──■──┤ X ├──■────■──┤ X ├
+            │  └───┘  │  ┌─┴─┐└───┘
+    qr_2: ──┼─────────■──┤ X ├─────
+          ┌─┴─┐     ┌─┴─┐└───┘
+    qr_3: ┤ X ├─────┤ X ├──────────
+          └───┘     └───┘
+    cr: 2/═════════════════════════
+    Args:
+        input_1: input 1 for the circuit.
+        input_2: input 2 for the circuit.
+        carry_in: carry in for the circuit.
+    Returns:
+        qiskit.result.counts.Counts: sum result counts.
+    >>> quantum_full_adder(1,1,1)
+    {'11': 1000}
+    >>> quantum_full_adder(0,0,1)
+    {'01': 1000}
+    >>> quantum_full_adder(1,0,1)
+    {'10': 1000}
```

Comment:
Please add some tests that fail...  Negative numbers, floating point numbers, letters, etc.

Reply:
Done!